openassessment/xblock/ui_mixins/mfe/mixin.py

```diff
@@ -62,61 +63,49 @@ def get_block_info(self, data, suffix=""):  # pylint: disable=unused-argument
     @XBlock.json_handler
     def get_learner_data(self, data, suffix=""):  # pylint: disable=unused-argument
         """
-        Get data for the user / step of the ORA
-        - If no step provided, go to current workflow step
-            - If not submitted, load draft submission
-            - If submitted, load fir
-        - If user provides jumpable step
-            - Check if we can get to that step
-            - If we can, go to that step
-            - Otherwise, raise an error
+        Get data for the user / step of the ORA, based on the following modes:
+
+        1) If no step provided, refresh progress but don't return any response.
+        2) If step provided, validate that we can get to that step and return the appropriate response
         """
+        # Query workflow step here only once to avoid duplicate workflow updates
         workflow_step = self.workflow_data.status or "submission"
```

Comment:
nit: variable names? 

workflow_step / active_step are very similar and easy to get mixed up which is which. maybe current_workflow_step and requested_step?